starter/src/main/java/com/vivy/springdoc/OpenAPIAutoConfiguration.java

```diff
@@ -32,4 +50,66 @@ public OpenAPI bearerOpenAPI() {
                 ))
                 .addSecurityItem(new SecurityRequirement().addList("bearer-jwt", List.of("read", "write")));
     }
+
+    @Bean
+    @ConditionalOnBean(OpenAPI.class)
+    public ApplicationRunner applicationRunner(Environment environment) {
+
+        String APPLICATION_PROPERTIES_PREFIX = "openapi-boot-starter.swagger-ui.api-docs.";
+        DefaultUriBuilderFactory uriFactory = new DefaultUriBuilderFactory();
+
+        return new ApplicationRunner() {
+            @Override
+            public void run(ApplicationArguments args) throws Exception {
+                int apiDocPort = environment.getProperty("local.server.port", Integer.class, -1);
```

Comment:
please bind all properties to an object

https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties

Reply:
thanks for the link, already looks better